cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerModificationService.java

```diff
@@ -575,12 +581,15 @@ private void doRestartServicesIfNeeded(ClustersResourceApi clustersResourceApi,
             ApiRestartClusterArgs restartClusterArgs = new ApiRestartClusterArgs();
             restartClusterArgs.setRedeployClientConfiguration(true);
             restartCommand = clustersResourceApi.restartCommand(stack.getName(), restartClusterArgs);
+            eventService
+                    .fireCloudbreakEvent(stack.getId(), UPDATE_IN_PROGRESS.name(), CLUSTER_CM_CLUSTER_SERVICES_RESTARTING);
         }
         if (restartCommand != null) {
             ExtendedPollingResult pollingResult = clouderaManagerPollingServiceProvider.startPollingCmServicesRestart(stack, apiClient, restartCommand.getId());
             handlePollingResult(pollingResult, "Cluster was terminated while restarting services.",
                     "Timeout happened while restarting services.");
         }
+        return restartCommand == null ? 0 : restartCommand.getId().intValue();
```

Comment:
This logic is also used in the `startServices()` Can you please extend to a private method and use it in both places?